server/routerlicious/packages/services-shared/src/storage.ts

```diff
@@ -41,7 +41,13 @@ export class DocumentStorage implements IDocumentStorage {
      */
     public async getDocument(tenantId: string, documentId: string): Promise<any> {
         const collection = await this.databaseManager.getDocumentCollection();
-        return collection.findOne({ documentId, tenantId });
+        const document = await collection.findOne({ documentId, tenantId });
+
+        if (document && document.deletionTime) {
+            return Promise.reject(new Error("Cannot retrieve deleted document."));
+        }
+
+        return document;
```

Comment:
No need to reject promise here. Caller (deli and scribe) is already checking the flag

Reply:
Deli and Scribe are both accessing Documents by directly calling the db ([deli](https://github.com/TriDucNgo/FluidFramework/blob/77cf4ef89a40102f520f8e04c58273eef3c492c7/server/routerlicious/packages/lambdas/src/deli/lambdaFactory.ts#L82), [scribe](https://github.com/TriDucNgo/FluidFramework/blob/77cf4ef89a40102f520f8e04c58273eef3c492c7/server/routerlicious/packages/lambdas/src/scribe/lambdaFactory.ts#L89-L92)). Looks like today only Alfred is using this functionality and would expect an error like accessing a deleted document to be surfaced up as a rejection.
1. [helper used for document PATCH calls](https://github.com/TriDucNgo/FluidFramework/blob/77cf4ef89a40102f520f8e04c58273eef3c492c7/server/routerlicious/packages/routerlicious-base/src/alfred/routes/api/api.ts#L184-L191)
2. [Document GET calls](https://github.com/TriDucNgo/FluidFramework/blob/77cf4ef89a40102f520f8e04c58273eef3c492c7/server/routerlicious/packages/routerlicious-base/src/alfred/routes/api/documents.ts#L42-L44)

Reply:
I think its better if the lower level APIs such as "getDocument" does not assume anything. They just return whatever you query. And the caller will be responsible for interpreting the result.

Reply:
Updated to remove check within `getDocuments(...)` and have callers perform the check.